jobbergate-cli/jobbergate_cli/subapps/job_scripts/app.py

```diff
@@ -244,21 +265,69 @@ def render(
         title="Created Job Script",
     )
 
-    # `submit` will be `None` --submit/--no-submit flag was not set
-    if submit is None:
-        # If not running in "fast" mode, ask the user what to do.
-        if not fast:
-            submit = typer.confirm("Would you like to submit this job immediately?")
+    # `submit` will be `None` when --submit/--no-submit flag was not set
+    if submit is None and (cluster_name or execution_directory):
+        # If the user specified --cluster-name or --execution-directory, they must also specify --submit
+        raise Abort(
+            "You must specify --cluster-name and --execution-directory only on submit mode.",
+            subject="Incorrect parameters",
+            support=True,
+        )
 
-        # Otherwise, assume that the job script should be submitted immediately
-        else:
-            submit = True
+    """
+    Invoke `question_helper` to handle the logic for parameters that are not specified by the user.
+
+    If the user specified --fast, the `question_helper` will not invoke the function and will
+    instead return the default answer. If the user did not specify --fast, the `question_helper`
+    will invoke the function and return the user's answer.
+    """
+
+    submit = question_helper(
+        question_func=typer.confirm,
+        text="Would you like to submit this job immediately?",
+        default=True,
+        fast=fast,
+        actual_value=submit,
+    )
+    cluster_name = question_helper(
+        question_func=typer.prompt,
+        text="What cluster should this job be submitted to?",
+        default=settings.DEFAULT_CLUSTER_NAME,
+        fast=fast,
+        actual_value=cluster_name,
+    )
+    execution_directory = question_helper(
+        question_func=typer.prompt,
+        text="Where should this job be executed?",
+        default=pathlib.Path.cwd(),
+        fast=fast,
+        actual_value=execution_directory,
+    )
+
+    download = question_helper(
+        question_func=typer.confirm,
+        text="Would you like to download the job script files?",
+        default=True,
+        fast=fast,
+        actual_value=download,
+    )
 
     if not submit:
         return
 
+    if download:
```

Comment:
This could go before
```
    if not submit:
         return
```

Reply:
Oh, same point I mentioned above.
I'll move it.